### PR TITLE
ADD application level authentication (with Cognito)

### PR DESCRIPTION
**Pull request by ramesh-maddegoda**
## 🗒️ Summary
Added application level authentication (with Cognito) to implement OAuth2 Authorization Code Grant with PKCE (Proof Key for Code Exchange) using react-oauth2-pkce library. 
Updated error messages to show relevant connectivity/auth related messages

## ♻️ Related Issues
Refer to the issue: NASA-PDS/doi-ui#159

### CODE REVIEW

File: src/AuthenticationWrapper.js [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+        const logoutUrl = OAUTH2_LOGOUT_ENDPOINT +
+            "?client_id=" + OAUTH2_CLIENT_ID +
+            "&logout_uri=" + OAUTH2_REDIRECT_URI;
+        window.location.replace(logoutUrl);
+    }
+
+    // If the user authentication is pending,show a Reset button to logout
+    if (authService.isPending()) {
+        return (<div align="center">
+            <h4>Authenticating...</h4>
```
[the remark this answers is not in the document]
Comment (jordanpadams) [reply src/AuthenticationWrapper.js:111]: @ramesh-maddegoda this one appears to still be associated with the latest code base. is this relevant still?

**Review by jordanpadams — CHANGES_REQUESTED** [review]

File: src/AuthenticationWrapper.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (ramesh-maddegoda) [reply src/AuthenticationWrapper.js:111]: @jordanpadams , I fixed the earlier error, where I did not use brackets after return and did not have a semi colon. After fixing it, it seems now sonartype think it as a regular expression. In this case it is allowed to have HTML defined in the  JavaScript. However, it seems sonar did not expect to see HTML here and analyzes it as a regular expression in JavaScript.

[the remark this answers is not in the document]
Comment (jordanpadams) [reply src/AuthenticationWrapper.js:111]: @ramesh-maddegoda copy. that is fine. feel free to use the sonatype commands in the future to close these out

[the remark this answers is not in the document]
Comment (jordanpadams) [reply src/AuthenticationWrapper.js:111]: @sonatype-lift ignore

[the remark this answers is not in the document]
Comment (jordanpadams) [reply src/AuthenticationWrapper.js:111]: @sonatype-lift ignore

File: package.json [hunk 2]
```diff
@@ -13,13 +13,16 @@
...
+    "jwt-decode": "^3.1.2",
     "react": "^17.0.2",
     "react-dom": "^17.0.2",
+    "react-oauth2-pkce": "^2.0.7",
     "react-redux": "^7.2.4",
     "react-router-dom": "^5.2.0",
     "react-scripts": "4.0.3",
     "redux": "^4.1.0",
     "redux-saga": "^1.1.3",
+    "styled-components": "^5.3.5",
```
[the remark this answers is not in the document]
Comment (jordanpadams) [reply package.json:25]: @sonatype-lift ignore

awaiting dependency upgrade to fix

**Review by jordanpadams — APPROVED** [review]

**Comment by ramesh-maddegoda** [comment]
Merged changes

**Pull request merged**